src/binfilereader.rs

```diff
@@ -33,24 +34,32 @@ macro_rules! bytes_to_primitive {
 pub struct BinFileReader {
     file_ptr: File,
     map: Mmap,
-    file_path: String,
+    file_path: PathBuf,
```

Comment:
Noted as a switch from `String` to `PathBuf`, not `&str`. There are some other interfaces that I could do this to as well.

Reply:
I think it's fine for it to just explode, from the POV of a consumer of the library if the docs told me it fails given invalid pixel values that'd be enough. 

Your rust is good, Errors are always hard, there's lots of good toys with iterators to make them more approachable for this sort of thing, but, I'd say that they're worth trying to sink ones teeth into first, focusing on how to handle errors later.